engine/packages/pegboard/src/workflows/serverless/pool.rs

```diff
@@ -0,0 +1,195 @@
+use futures_util::FutureExt;
+use gas::{db::WorkflowData, prelude::*};
+use rivet_types::{keys, runner_configs::RunnerConfigKind};
+
+use super::runner;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Input {
+	pub namespace_id: Id,
+	pub runner_name: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Default)]
+struct LifecycleState {
+	runners: Vec<Id>,
+}
+
+#[workflow]
+pub async fn pegboard_serverless_pool(ctx: &mut WorkflowCtx, input: &Input) -> Result<()> {
+	ctx.loope(LifecycleState::default(), |ctx, state| {
+		let input = input.clone();
+		async move {
+			// 1. Remove completed connections
+			let completed_runners = ctx
+				.activity(GetCompletedInput {
+					runners: state.runners.clone(),
+				})
+				.await?;
+
+			state.runners.retain(|r| !completed_runners.contains(r));
+
+			// 2. Get desired count -> drain and start counts
+			let ReadDesiredOutput::Desired(desired_count) = ctx
+				.activity(ReadDesiredInput {
+					namespace_id: input.namespace_id,
+					runner_name: input.runner_name.clone(),
+				})
+				.await?
+			else {
+				return Ok(Loop::Break(()));
+			};
+
+			let drain_count = state.runners.len().saturating_sub(desired_count);
+			let start_count = desired_count.saturating_sub(state.runners.len());
+
+			// 3. Drain old runners
+			if drain_count != 0 {
+				// TODO: Implement smart logic of draining runners with the lowest allocated actors
```

Comment:
note this doesn't matter rn since all major platforms use 1 actor per runner, but will be needed when using non-serverless